README.md

```diff
@@ -0,0 +1,123 @@
+
+The project is organized into several key components, each responsible for specific functionalities:
+
+### Controllers
+
+- **AuthController**: Manages user-related operations such as registration, login, and profile management.
+- **BookController**: Manages book processing, including book creation and updating for admins, and viewing for users.
+- **OrderController**: Manages order processing, including order creation, updating, and viewing order history.
+- **CategoryController**: Manages category processing, including category creation, updating, and viewing.
+- **ShoppingCartController**: Manages shopping cart processing, including creation, updating, and viewing items (books and other products).
+
+### Services
+
+- **OrderService**: Manages the order processing logic, including placing orders in the database, retrieving order history, and calculating the total price of all items.
+- **BookService**: Manages the book processing logic, including book management for admins and book retrieval for users.
+- **CategoryService**: Manages category processing logic, including category management for admins and category retrieval for users.
+- **ShoppingCartService**: Handles the business logic related to shopping carts, including adding items and updating the cart.
+- **UserService**: Contains business logic for user operations such as registration and user retrieval by username (email).
+
+### Repositories
+
+- **BookRepository**: Interface for CRUD operations on Book entities and special methods for finding books by category.
+- **ShoppingCartRepository**: Interface for CRUD operations on ShoppingCart entities and methods for identifying a shopping cart by user ID.
+- **CartItemRepository**: Interface for CRUD operations on CartItem entities.
+- **OrderRepository**: Interface for CRUD operations on Order entities.
+- **OrderItemRepository**: Interface for CRUD operations on OrderItem entities and methods for retrieving order items by order ID.
+- **UserRepository**: Interface for CRUD operations on User entities and methods for checking if the user is already registered.
+- **RoleRepository**: Interface for CRUD operations on Role entities, including methods for identifying user roles.
+
+## Setup and Usage
+
+To get started with this project, follow these steps:
+
+1. **Clone the repository**:
+    ```sh
+    git clone https://github.com/bodya4243/online-book-store.git
+    cd online-book-store
+    ```
+
+2. **Build the project using Maven**:
+    ```sh
+    mvn clean install
+    ```
+
+3. **Run the application**:
+    ```sh
+    mvn spring-boot:run
+    ```
+
+4. **Access the application**:
+    - The application runs on `http://localhost:8080`.
+    - API documentation is available at `http://localhost:8080/swagger-ui.html`.
+
+5. **Swagger**:
+    - You can use Swagger for manipulating API endpoints.
+
+6. **Docker**:
+    - You can download the container from Docker and start the API.
+
+### Docker Setup
```

Comment:
add the auth and usage guide somewhere after the project initialization topic. 
Mention how it works (JWT token) where can the user get it, how can he register, and also add admin login & password so that the user of your app can test admin-only endpoints as well. It's a demonstration project after all, so we can add admin credentials here

We should also add the link to the deployed project when and if you will deploy it